Error relating to jsonld values in list format

After playing around with the JSON-LD metadata produced by the [doomlab data dictionary creator](https://doomlab.shinyapps.io/ddcreator/), I tried using it as the dataset_description.json file for a toy dataset, and it produced the following error:

```
        ******************************************************
        This dataset does not appear to be psych-DS compatible
        ******************************************************

	[ERROR] There is a column name in your CSV data that is not listed in the "variableMeasured" property in your metadata file. The missing column names are shown below. (CSV_COLUMN_MISSING_FROM_METADATA)

		./data/framedata-per-response/study-a03dcccc_response-26c4c767_data.csv
			Evidence: Column headers: [response_uuid,child_hashed_id,frame_id,event_number,key,value] do not appear in variableMeasured. schema.rules.csv_data.Datafile

		1 more files with the same issue

	[ERROR] Your metadata is missing the required schema.org "Dataset" type. If you are having trouble creating your dataset_description.json file, you can download a template by filling out the form at https://psych-ds.github.io/cedar-wizard-psychds/. (INCORRECT_DATASET_TYPE)

		./dataset_description.json
			Evidence: dataset_description.json's "@type" property must have "Dataset" as its value.
                      additionally, the term "Dataset" must implicitly or explicitly use the schema.org namespace.
                      The schema.org namespace can be explicitly set using the "@context" key

		1 more files with the same issue

	[ERROR] The metadata object listed below is missing a required key. (JSON_KEY_REQUIRED)

		./data/framedata-per-response/study-a03dcccc_response-26c4c767_data.csv
			Evidence: metadata object missing fields: [name,description,variableMeasured] as per schema.rules.compiled_metadata.CompiledMetadata. 
                    If these fields appear to be present in your metadata, then there may be an issue with your schema.org context

		1 more files with the same issue


There was an issue with your variableMeasured value. Here is a suggested value:

["response_uuid","child_hashed_id","frame_id","event_number","key","value"]
```

This was confusing, because the metadata file definitely had the correct fields and variableMeasured values. I suspect that on some level, the issue is being caused by the fact that the doomlab output JSON uses singleton arrays (e.g. `"@type": ["Dataset"]`) rather than bare strings. 

When I changed the "@context" and "@type" fields to bare strings in the toy dataset, there was a different, uncaught error:

```
An error occurred: TypeError: Cannot read properties of undefined (reading 'includes')
    at _schemaCheck (file:///Users/brianleonard/psych-ds/clean/psychds-validator/src/schema/applyRules.ts:492:31)
    at _schemaCheck (file:///Users/brianleonard/psych-ds/clean/psychds-validator/src/schema/applyRules.ts:513:26)
    at _schemaCheck (file:///Users/brianleonard/psych-ds/clean/psychds-validator/src/schema/applyRules.ts:513:26)
    at schemaCheck (file:///Users/brianleonard/psych-ds/clean/psychds-validator/src/schema/applyRules.ts:299:12)
    at Object.evalColumns [as columnsMatchMetadata] (file:///Users/brianleonard/psych-ds/clean/psychds-validator/src/schema/applyRules.ts:198:3)
    at file:///Users/brianleonard/psych-ds/clean/psychds-validator/src/schema/applyRules.ts:142:19
    at Array.map (<anonymous>)
    at evalRule (file:///Users/brianleonard/psych-ds/clean/psychds-validator/src/schema/applyRules.ts:140:6)
    at applyRules (file:///Users/brianleonard/psych-ds/clean/psychds-validator/src/schema/applyRules.ts:46:7)
    at applyRules (file:///Users/brianleonard/psych-ds/clean/psychds-validator/src/schema/applyRules.ts:53:7)
```

This indicates that somewhere within the schemaCheck function, the superClassSlots array is becoming undefined due to the changes we made.

I think going down the rabbit hole of this traceback will be very informative. Even though the singleton array/bare string difference is attested for every little field in the metadata, I think in the end that the answer will specifically pertain to how "@type" and "@context" fields are handled. You'll notice that for simple text fields like "name" and "description", the code never even really tries to access the value at any point, so the difference in format doesn't matter. I believe that two fixes will be required, one pertaining to how "@type" is handled during schemaCheck in applyRules.ts, and one pertaining to how "@context" is handled during getExpandedSidecar in context.ts. 

Here's the relevant chunk in context.ts, where all the different possible URL formats for schema.org are accounted for, and any use of a bare string URL in "@context" is converted to the more stable and reliable "@vocab" format (which is necessary for getting the expansion function to work properly in a browser context, for some reason.)

These errors came up while using the comments_and_format branch, which is currently up as a PR. 

You can run the validator like this. Make sure your deno is up to date (brew upgrade deno):
```
deno run -A src/index.ts <testdata dir>
```
I mentioned using the npm version earlier, but I forgot that you'll be debugging, so working in the deno context is better for sure.

Here's the toy dataset we were using:
[testdata.zip](https://github.com/user-attachments/files/18367780/testdata.zip)

Valid datasets are stored in test_data/valid_datasets and can be used for comparison 

I'm sure there's more things that I could add to this, and I'll make sure to do that when I think of them.

--- a/src/schema/applyRules.ts
+++ b/src/schema/applyRules.ts
@@ -415,7 +415,7 @@ import { psychDSFile } from '../types/file.ts';
           //TODO: add else statement? if property is not a valid schema.org slot at all
           let subKeys: string[] = []
           //if current property is not a valid slot of this object type, raise issue
-          if (!(superClassSlots.includes(property))){
+          if (!superClassSlots.includes(property)){
             issues.termIssues.push(`${objectPath}.${property}`)
           }
           else{
@@ -460,25 +460,17 @@ import { psychDSFile } from '../types/file.ts';
     nameSpace: string
   ): string[]{
     
-    if(type.includes(nameSpace)){
-      type = type.replace(nameSpace,"")
-    }
-    if(type in schema[`schemaOrg.classes`]){
-      //if type has a super class, append this type's slots to the result of this function for super class
-      if('is_a' in schema[`schemaOrg.classes.${type}`]){
-        if('slots' in schema[`schemaOrg.classes.${type}`]){
-          return (schema[`schemaOrg.classes.${type}.slots`] as unknown as string[]).concat(getSuperClassSlots(schema[`schemaOrg.classes.${type}.is_a`] as unknown as string,schema,nameSpace))
+    type = type.replace(nameSpace,"")
 
-        }
-        else
-          return getSuperClassSlots(schema[`schemaOrg.classes.${type}.is_a`] as unknown as string,schema,nameSpace)
-      }
-      //TODO: shouldn't another if statement to check for slots presence be needed here?
-      else
-        return schema[`schemaOrg.classes.${type}.slots`] as unknown as string[]
-
+    if (!(type in schema['schemaOrg.classes'])) {
+      return []
     }
-    return []
+
+    //if type has a super class, append this type's slots to the result of this function for super class
+    const slots = schema[`schemaOrg.classes.${type}.slots`] as string[] || []
+    const is_a = 'is_a' in schema[`schemaOrg.classes.${type}`] ? getSuperClassSlots(schema[`schemaOrg.classes.${type}.is_a`] as string, schema, nameSpace) : []
+
+    return [...slots, ...is_a];
   }
 
   //recursive function for finding all classes that are more specific versions of a given class

--- a/src/schema/context.ts
+++ b/src/schema/context.ts
@@ -345,10 +345,19 @@ import {
         //in addition to adding the appropriate namespace (e.g. http://schema.org)
         //to all keys within the json, it also throws a variety of errors for improper JSON-LD syntax,
         //which mostly all pertain to improper usages of privileged @____ keywords
-        if ('@context' in this.sidecar && typeof this.sidecar['@context'] == 'string' && ['http://schema.org/','http://schema.org','http://www.schema.org/','http://www.schema.org','https://schema.org/','https://schema.org','https://www.schema.org/','https://www.schema.org/'].includes(this.sidecar['@context']))
-        this.sidecar['@context'] = {
-          '@vocab':'http://schema.org/'
+        if ('@context' in this.sidecar){
+          
+          if (Array.isArray(this.sidecar['@context']) && this.sidecar['@context'].length === 1){
+            this.sidecar['@context'] = this.sidecar['@context'][0]
+          }
+
+          if (typeof this.sidecar['@context'] == 'string' && ['http://schema.org/','http://schema.org','http://www.schema.org/','http://www.schema.org','https://schema.org/','https://schema.org','https://www.schema.org/','https://www.schema.org/'].includes(this.sidecar['@context'])){
+            this.sidecar['@context'] = {
+              '@vocab':'http://schema.org/'
+            }
+          }
         }
+
         const exp = await jsonldToUse.expand(this.sidecar,expandOptions)
         return exp[0] || {};
       }

